internal/kafka/internal/api/dbapi/kafka_request_types.go

```diff
@@ -140,3 +140,48 @@ func (k *KafkaRequest) GetExpirationTime(lifespanSeconds int) *time.Time {
 	expireTime := k.CreatedAt.Add(time.Duration(lifespanSeconds) * time.Second)
 	return &expireTime
 }
+
+type LifeSpan interface {
+	Infinite() bool
+	Days() float64
+}
+
+var _ LifeSpan = &lifeSpan{}
+
+type lifeSpan struct {
+	lifeSpanDays float64
+	infinite     bool
+}
+
+func (l *lifeSpan) Infinite() bool {
+	return l.infinite
+}
+
+func (l *lifeSpan) Days() float64 {
+	return l.lifeSpanDays
+}
+
+// IsExpired returns whether a kafka is expired and how many days the instance will live before expiring (-1 never expires)
+func (k *KafkaRequest) IsExpired() (bool, LifeSpan) {
+	// TODO: change this when the NullTime will be used
+	if k.ExpiresAt.IsZero() {
+		return false, &lifeSpan{
+			lifeSpanDays: -1,
+			infinite:     true,
+		}
+	}
+
+	now := time.Now()
+	if now.After(k.ExpiresAt) {
+		return true, &lifeSpan{
+			lifeSpanDays: 0,
+			infinite:     false,
+		}
+	}
+
+	remainingLife := k.ExpiresAt.Sub(now)
+	return false, &lifeSpan{
+		lifeSpanDays: remainingLife.Hours() / 24,
```

Comment:
This will result in inaccuracies of up to an hour.
A kafka instance that expires in 59 minutes will return a lifeSpanDays of 0 however it will return that it is not expired (false) at the same time.